pytrickle/stream_processor.py

```diff
@@ -319,8 +383,36 @@ async def process_audio_async(self, frame: AudioFrame) -> Optional[List[AudioFra
             return [frame]
     
     async def update_params(self, params: Dict[str, Any]):
-        """Update parameters using provided async function."""
-        if self.param_updater:
+        """Update parameters using provided async function.
+        
+        Handles the 'load_model' sentinel parameter internally by extracting it
+        and triggering model loading, while passing through all other parameters
+        to the user's param_updater callback.
+        """
+        # Handle model loading sentinel message (extract it with pop)
+        if params.pop("load_model", False):
+            try:
+                await self.ensure_model_loaded()
+                logger.info(f"Model loaded via parameter update for '{self.name}'")
+                # Fall through to process remaining params (if any)
+            except Exception as e:
+                logger.error(f"Error loading model via parameter update: {e}")
+                raise
+
+        # Handle warmup sentinel message (extract it with pop)
+        warmup_params = params.pop("warmup", None)
+        if warmup_params is not None:
```

Comment:
Should this trigger on all model loads as well?

Reply:
I simplified this logic a bit, 

https://github.com/livepeer/pytrickle/pull/36/commits/71ecd121839792ff21b8309a38153101cd33df66